tests/acceptance/30_custom_promise_types/24_expireafter.cf

```diff
@@ -0,0 +1,86 @@
+######################################################
+#
+#  Test that expireafter works with custom promise types
+#
+#####################################################
+body common control
+{
+  inputs => { "../default.cf.sub" };
+  bundlesequence  => { default("$(this.promise_filename)") };
+  version => "1.0";
+}
+
+#######################################################
+
+bundle common version_check
+{
+  classes:
+      "python_version_compatible_with_cfengine_library"
+        expression => returnszero(
+          "/usr/bin/python3 -c 'import sys; assert sys.version_info >= (3,6)'",
+          "useshell"
+        );
+
+@if minimum_version(3.20)
+      "custom_promises_supports_expireafter"
+        expression => "any";
+@endif
+}
+
+bundle agent init
+{
+  meta:
+      "test_skip_unsupported"
+        string => "!custom_promises_supports_expireafter|!python_version_compatible_with_cfengine_library";
+
+  files:
+      "$(this.promise_dirname)/cfengine.py"
+        copy_from => local_cp(
+          "$(this.promise_dirname)/../../../modules/promises/cfengine.py"
+        );
+      "/tmp/testfile.txt"
+        delete => tidy;
+}
+
+#######################################################
+
+body action background {
+  background => "true";
+}
+
+bundle agent test
+{
+  meta:
+      "description" -> { "CFE-3435" }
+        string => "Test that expireafter works with custom promise types";
+
+  commands:
+      "$(sys.cf_agent) --inform --file $(this.promise_filename).sub --"
+        action => background;
+      "/bin/sleep 90";
+      "$(sys.cf_agent) --inform --file $(this.promise_filename).sub";
+      "/bin/sleep 100";
+}
+
+#######################################################
+
+bundle agent check
+{
+  vars:
+      "expected"
+        string => "Hello CFEngine!";
+      "actual"
+        string => readfile("/tmp/testfile.txt"),
+        if => fileexists("/tmp/testfile.txt");
+  classes:
+      "ok"
+        expression => strcmp("$(expected)", "$(actual)");
```

Comment:
I guess I am all turned around then.

I thought that the `slow_append` promise was suppose to not be allowed to finish writing (because it would be killed by the next cf-agent process hitting that promise). But here the test passes if the slowly written file has the expected content. I thought this would test that the `slow_append` promise was properly terminated and that it didn't finish writing the file.

Reply:
If the first agent is not killed, we will read `Hello CFEngine!Hello CFEngine!` and test will fail. If the first agent is killed, we will read `Hello CFEngine!` and test will pass.